File: LLM-Translations/GPT-4/avatar/Python/Go/atcoder_AGC034_B.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func gcd(a, b int) int {
	a, b = max(a, b), min(a, b)
	for a%b > 0 {
		a, b = b, a%b
	}
	return b
}

func solve() {
	reader := bufio.NewReader(os.Stdin)
	s, _ := reader.ReadString('\n')
	s = strings.TrimSuffix(s, "\n")
	t := ""
	i := 0
	for i < len(s) {
		if s[i] == 'A' {
			t += "A"
			i++
		} else if s[i] == 'B' {
			if i < len(s)-1 {
				if s[i+1] == 'C' {
					t += "D"
					i += 2
				} else {
					t += "X"
					i++
				}
			} else {
				t += "X"
				i++
			}
		} else {
			t += "X"
			i++
		}
	}
	total := 0
	numA := 0
	for i := range t {
		if t[i] == 'X' {
			numA = 0
		} else {
			if t[i] == 'A' {
				numA++
			} else {
				total += numA
			}
		}
	}
	fmt.Println(total)
}

func main() {
	solve()
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/atcoder_AGC034_B.java

import java.util.*;

public class atcoder_AGC034_B {
    public static int gcd(int a, int b) {
        a = Math.max(a, b);
        b = Math.min(a, b);
        while (a % b > 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return b;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        String t = "";
        int i = 0;
        while (i < s.length()) {
            if (s.charAt(i) == 'A') {
                t += "A";
                i++;
            } else if (s.charAt(i) == 'B') {
                if (i < s.length() - 1) {
                    if (s.charAt(i + 1) == 'C') {
                        t += "D";
                        i += 2;
                    } else {
                        t += "X";
                        i++;
                    }
                } else {
                    t += "X";
                    i++;
                }
            } else {
                t += "X";
                i++;
            }
        }
        int total = 0;
        int numA = 0;
        for (i = 0; i < t.length(); i++) {
            if (t.charAt(i) == 'X') {
                numA = 0;
            } else {
                if (t.charAt(i) == 'A') {
                    numA++;
                } else {
                    total += numA;
                }
            }
        }
        System.out.println(total);
        final long INF = (long) Math.pow(10, 25);
        final long mod = (long) (7 + Math.pow(10, 9));
    }
}

End of Code
